File: models/Transaction.js
const mongoose = require('mongoose');

const transactionSchema = new mongoose.Schema({
    farmerId: { type: mongoose.Schema.Types.ObjectId, ref: 'Farmer', required: true },
    type: { type: String, enum: ['payment', 'advance', 'collection'], required: true },
    amount: { type: Number, required: true },
    mode: { type: String, enum: ['cash', 'bank', 'upi', 'system'], default: 'cash' }, // 'system' for auto-deductions etc
    date: { type: Date, default: Date.now },
    description: { type: String }
});

module.exports = mongoose.model('Transaction', transactionSchema);

File: cloud_dairy/lib/models/transaction.dart
class Transaction {
  final String id;
  final String farmerId;
  final String type; // payment, advance, collection
  final double amount;
  final String mode;
  final String description;
  final DateTime date;

  Transaction({
    required this.id,
    required this.farmerId,
    required this.type,
    required this.amount,
    required this.mode,
    required this.description,
    required this.date,
  });

  factory Transaction.fromJson(Map<String, dynamic> json) {
    return Transaction(
      id: json['_id'],
      farmerId: json['farmerId'],
      type: json['type'],
      amount: (json['amount'] ?? 0).toDouble(),
      mode: json['mode'] ?? '',
      description: json['description'] ?? '',
      date: DateTime.parse(json['date']),
    );
  }
}
